pyccl/nl_pt/ept.py

```diff
@@ -0,0 +1,700 @@
+        """
+        self._check_pt_init()
+        # Get Pk templates
+        Pd1d1 = self.pk_b1
+        Pd1d2 = self._g4[:, None] * self.dd_bias[2][None, :]
+        Pd2d2 = self._g4[:, None] * self.dd_bias[3][None, :]
+        Pd1s2 = self._g4[:, None] * self.dd_bias[4][None, :]
+        Pd2s2 = self._g4[:, None] * self.dd_bias[5][None, :]
+        Ps2s2 = self._g4[:, None] * self.dd_bias[6][None, :]
+        Pd1p3 = self._g4[:, None] * self.dd_bias[8][None, :]
+        Pd1k2 = self.pk_bk * (self.k_s**2)[None, :]
+
+        # Get biases
+        b11 = tr1.b1(self.z_s)
+        b21 = tr1.b2(self.z_s)
+        bs1 = tr1.bs(self.z_s)
+        bk21 = tr1.bk2(self.z_s)
+        b3nl1 = tr1.b3nl(self.z_s)
+        b12 = tr2.b1(self.z_s)
+        b22 = tr2.b2(self.z_s)
+        bs2 = tr2.bs(self.z_s)
+        bk22 = tr2.bk2(self.z_s)
+        b3nl2 = tr2.b3nl(self.z_s)
+
+        s4 = 0.
+        if self.fastpt_par['sub_lowk']:
+            s4 = self._g4 * self.dd_bias[7]
+            s4 = s4[:, None]
+
+        pgg = ((b11*b12)[:, None] * Pd1d1 +
+               0.5*(b11*b22 + b12*b21)[:, None] * Pd1d2 +
+               0.25*(b21*b22)[:, None] * (Pd2d2 - 2.*s4) +
+               0.5*(b11*bs2 + b12*bs1)[:, None] * Pd1s2 +
+               0.25*(b21*bs2 + b22*bs1)[:, None] * (Pd2s2 - (4./3.)*s4) +
+               0.25*(bs1*bs2)[:, None] * (Ps2s2 - (8./9.)*s4) +
+               0.5*(b12*b3nl1+b11*b3nl2)[:, None] * Pd1p3 +
+               0.5*(b12*bk21+b11*bk22)[:, None] * Pd1k2)
+
+        return pgg*self.exp_cutoff
+
+    def _get_pgi(self, trg, tri):
+        """ Get the number counts - IA cross-spectrum at the internal
+        set of wavenumbers and scale factors.
+
+        .. note:: The full non-linear model for the cross-correlation
+                  between number counts and intrinsic alignments is
+                  still work in progress in FastPT. As a workaround
+                  CCL assumes a non-linear treatment of IAs, but only
+                  linearly biased number counts.
+
+        Args:
+            trg (:class:`~pyccl.nl_pt.tracers.PTTracer`): number
+                counts tracer.
+            tri (:class:`~pyccl.nl_pt.tracers.PTTracer`): intrinsic
+                alignment tracer.
+
+        Returns:
+            array_like: 2D array of shape `(N_a, N_k)`, where `N_k` \
+                is the size of this object's `k_s` attribute, and \
+                `N_a` is the size of the object's `a_s` attribute.
+        """
+        self._check_pt_init()
+        # Get Pk templates
+        Pd1d1 = self.pk_b1
+        a00e, c00e, a0e0e, a0b0b = self.ia_ta
+        a0e2, b0e2, d0ee2, d0bb2 = self.ia_mix
+
+        # Get biases
+        b1 = trg.b1(self.z_s)
+        b2 = trg.b2(self.z_s)
+        bs = trg.bs(self.z_s)
+        bk2 = trg.bk2(self.z_s)
+        b3nl = trg.b3nl(self.z_s)
+        if (np.any(b2 != 0) or np.any(bs != 0) or
+                np.any(bk2 != 0) or np.any(b3nl != 0)):
```

Comment:
`if any([b.any() for b in [b2, bs, bk2, b3nl]])`

Reply:
I prefer to be explicit

Reply:
For explicit checks with 0 you can do `if any([(b != 0).any() for b in [b2, bs, bk2, b3nl]])`.
Also, the method `.any()` is considerably faster than the function `np.any()`, so you want to use it when you can avoid using the function. In this case, we know that these will be numpy arrays.

Reply:
Changed to your original suggestion.